### PR TITLE
fix(cascader): fix leaf nodes not closing the last expanded panel in hover state

### LINKED ISSUE #41134: Cascader组件在设置expandTrigger为hover时，如果悬停到没有下级菜单的一个一级菜单上，不会隐藏之前显示的其它一级菜单的二级菜单
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/basic-antd-5-3-0-forked-s8zx1f?file=/demo.tsx:1002-1015)

### Steps to reproduce

1. 鼠标hover到Zhejiang上，会显示Hangzhou二级菜单；
2. 鼠标hover到beijing上，由于beijing没有children，且设置了isLeaf为true，所以没有二级菜单，但是没有隐藏Hangzhou二级菜单；
3. expandTrigger属性设置为click(默认值)时，没有上述问题

### What is expected?

在设置expandTrigger为hover时，如果悬停到没有下级菜单的一个一级菜单上，会隐藏之前显示的其它一级菜单的二级菜单

### What is actually happening?

在设置expandTrigger为hover时，如果悬停到没有下级菜单的一个一级菜单上，不会隐藏之前显示的其它一级菜单的二级菜单

| Environment | Info |
| --- | --- |
| antd | 5.3.0 |
| React | 18.0.0 |
| System | Mac |
| Browser | Chrome 111 |

---
具体应该是https://github.com/react-component/cascader/blob/e7b6d156f6f3e303d4f7e1a81575a9abbb3be382/src/OptionList/Column.tsx中的

```javascript
const triggerOpenPath = () => {
            if (!disabled && (!hoverOpen || !isMergedLeaf)) {
              onActive(fullPath);
            }
};
```

方法中，对于leaf节点，限制了!hoverOpen时才生效。

### PR DESCRIPTION
**Pull request by Wxh16144**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix #41134 

### 💡 Background and solution

**Before**

https://user-images.githubusercontent.com/32004925/224211927-4301381b-8dde-42cf-a32b-b89304c58aad.mp4

**After**

https://user-images.githubusercontent.com/32004925/224211915-a06c7196-c855-4ea7-a55b-4fff9c804640.mp4

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     fix the bug of leaf nodes not closing the last expanded panel in hover state      |
| 🇨🇳 Chinese |    修复叶子节点在 hover 状态未关闭上一次展开的面板问题        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -120,7 +120,7 @@
     "copy-to-clipboard": "^3.2.0",
     "dayjs": "^1.11.1",
     "qrcode.react": "^3.1.0",
-    "rc-cascader": "~3.9.0",
+    "rc-cascader": "~3.9.1",
```
Comment (kiner-tang) [inline package.json:123]: 只是`patch`版本是不是不用单独在`antd`提`pr`?

Comment (kiner-tang) [reply package.json:123]: `patch`版本，用户只要重新安装依赖就会装最新的`3.9.1`了

Comment (zombieJ) [reply package.json:123]: 直接到这里更新日志吧：
https://github.com/ant-design/ant-design/pull/41190

Comment (Wxh16144) [reply package.json:123]: 我也在想， 但是没人回复我 QaQ

**Pull request closed without merging**